KerbalStuff/thumbnail.py

```diff
@@ -1,12 +1,48 @@
 import os.path
-from typing import Tuple
 
 from PIL import Image
 
+from KerbalStuff.config import _cfg, _cfgi, site_logger
 
-def create(imagePath: str, thumbnailPath: str, thumbnailSize: Tuple[int, int]) -> None:
-    if not os.path.isfile(imagePath):
+
+def create(background_path: str, thumbnail_path: str) -> None:
+    if not os.path.isfile(background_path):
         return
-    im = Image.open(imagePath)
-    im.thumbnail(thumbnailSize, Image.ANTIALIAS)
-    im.save(thumbnailPath, 'jpeg', quality=50, optimize=True)
+
+    size_str = _cfg('thumbnail_size')
+    if not size_str:
+        size_str = "320x320"
+    size_str_tuple = size_str.split('x')
+    size = (int(size_str_tuple[0]), int(size_str_tuple[1]))
+
+    quality = _cfgi('thumbnail_quality')
+    if not quality or quality > 100:
+        quality = 80
+
+    im = Image.open(background_path)
+    if im.mode in ("RGBA", "P"):
```

Comment:
I was curious about these modes, and apparently there are a lot of them:

https://pillow.readthedocs.io/en/5.1.x/handbook/concepts.html#modes

> - 1 (1-bit pixels, black and white, stored with one pixel per byte)
> - L (8-bit pixels, black and white)
> - P (8-bit pixels, mapped to any other mode using a color palette)
> - RGB (3x8-bit pixels, true color)
> - RGBA (4x8-bit pixels, true color with transparency mask)
> - CMYK (4x8-bit pixels, color separation)
> - YCbCr (3x8-bit pixels, color video format)
> - Note that this refers to the JPEG, and not the ITU-R BT.2020, standard
> - LAB (3x8-bit pixels, the L*a*b color space)
> - HSV (3x8-bit pixels, Hue, Saturation, Value color space)
> - I (32-bit signed integer pixels)
> - F (32-bit floating point pixels)
> 
> PIL also provides limited support for a few special modes, including LA (L with alpha), RGBX (true color with padding) and RGBa (true color with premultiplied alpha). However, PIL doesn’t support user-defined modes; if you need to handle band combinations that are not listed above, use a sequence of Image objects.

Should we think about converting _all_ of them to RGB?

Reply:
That makes sense :+1: